cli/azd/pkg/async/task_context.go

```diff
@@ -0,0 +1,97 @@
+//nolint:all
+package async
+
+// Task function definition
+type TaskRunFunc[R comparable] func(taskContext *TaskContext[R])
```

Comment:
Yes, I was attempting to keep it as simple as possible, however if we support the simpler return `(R, error)` then we would limit usage of go routines I believe.  

As an example take a look at the `Deploy` func in the `bicep_provider`.  This runs a go routine in addition to the standard task func.  The go routine is the one that sets the result value.

Reply:
That alone does not mean we can't use a simpler signature for task func. Taking the specific example you mentioned, instead of setting `isDeploymentComplete` flag, the deployment goroutine could pass the result to the task func via a channel, and the task func could then return the result.

Reply:
Going to keep the design as is for the moment since most implementations of the async task also support progress reporting and interactive console integration.

Reply:
😑